crates/bevy_ecs/macros/src/query_filter.rs

```diff
@@ -120,7 +120,8 @@ pub fn derive_query_filter_impl(input: TokenStream) -> TokenStream {
     );
 
     let filter_impl = quote! {
-        impl #user_impl_generics #path::query::QueryFilter
+        // SAFETY: This only performs access that subqueries perform, and they impl `QueryFilter` and so perform no mutable access.
+        unsafe impl #user_impl_generics #path::query::QueryFilter
```

Comment:
Would it be better to say that this `impl` inherits the safety properties of the subqueries, rather than assuming that they are correct? Unsure, personally.